controllers/managedfusion_controller.go

```diff
@@ -689,6 +723,123 @@ func (r *ManagedFusionReconciler) reconcilePrometheusProxyNetworkPolicy() error
 	return nil
 }
 
+func (r *ManagedFusionReconciler) reconcileEgressFirewall() error {
+	if !r.AvailableCRDs[EgressFirewallCRD] {
+		r.Log.Info("EgressFirewall CRD not present, skipping reconcile for egress firewall")
+		return nil
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressFirewall, func() error {
+		if err := r.own(r.egressFirewall); err != nil {
+			return err
+		}
+		desiredEgressSpec, err := r.getEgressDesiredState()
+		if err != nil {
+			return err
+		}
+		r.egressFirewall.Spec = *desiredEgressSpec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update egressFirewall: %v", err)
+	}
+	return nil
+
+}
+
+func (r *ManagedFusionReconciler) reconcileEgressNetworkPolicy() error {
+	if !r.AvailableCRDs[EgressNetworkPolicyCRD] {
+		r.Log.Info("EgressNetworkPolicy CRD not present, skipping reconcile for egress network policy")
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressNetworkPolicy, func() error {
+		if err := r.own(r.egressNetworkPolicy); err != nil {
+			return err
+		}
+		desiredEgressFirewallSpec, err := r.getEgressDesiredState()
+		if err != nil {
+			return err
+		}
+		var desiredEgressNetworkPolicy openshiftv1.EgressNetworkPolicy
+		for i := range desiredEgressFirewallSpec.Egress {
+			desiredEgressNetworkPolicy.Spec.Egress = append(
+				desiredEgressNetworkPolicy.Spec.Egress,
+				openshiftv1.EgressNetworkPolicyRule{
+					Type: openshiftv1.EgressNetworkPolicyRuleType(desiredEgressFirewallSpec.Egress[i].Type),
+					To:   openshiftv1.EgressNetworkPolicyPeer(desiredEgressFirewallSpec.Egress[i].To),
+				},
+			)
+		}
```

Comment:
The conversion here is problematic as it depends on implementation knowledge of the different structs and enum values

I would prefer to have an explicit conversion and I am suggesting the following util function (in the reconciler file, but not on the reconciler itself)

```
func egressFirewallSpecToEgressNetworkPolicySpec(firewallSpec *ovnv1.EgressFirewallSpec) *openshiftv1.EgressNetworkPolicySpec {
	networkPolicySpec := &openshiftv1.EgressNetworkPolicySpec{}

	for i := range firewallSpec.Egress {
		firewalRule := &firewallSpec.Egress[i]
		networkPolicyRule := openshiftv1.EgressNetworkPolicyRule{}

		// Convert rule type
		switch firewalRule.Type {
		case ovnv1.EgressFirewallRuleAllow:
			networkPolicyRule.Type = openshiftv1.EgressNetworkPolicyRuleAllow
		case ovnv1.EgressFirewallRuleDeny:
			networkPolicyRule.Type = openshiftv1.EgressNetworkPolicyRuleDeny
		}

		// Convert Rule Peer
		networkPolicyRule.To = openshiftv1.EgressNetworkPolicyPeer{
			CIDRSelector: firewalRule.To.CIDRSelector,
			DNSName:      firewalRule.To.DNSName,
		}

		networkPolicySpec.Egress = append(
			networkPolicySpec.Egress,
			networkPolicyRule,
		)
	}

	return networkPolicySpec
}

```

Reply:
Implemented the solution that @leelavg suggested